### ISSUE TITLE
API's request-transformer overriden by Consumer's request-transformer

### ISSUE THREAD

**Issue by dojusa** [Mashape/kong#2922]
I added an global request-transformer plugin for all my API to rename one request header. Everythings wokrs fine at this point.

But when I put a request-transformer to an specific consumer, it overrides the API request-transformer. The consumer's transformer only adds a new header, which in theory should affect the API's transformer.

Is there a reason for the behaviour? Would be nice to have both request-transformers working together.

**Comment by Tieske** [comment]
this is a design limitation, every plugin runs only once. And in this case the consumer one is the "more specific", and hence get priority.

This will likely change in the future.

PS. a workaround might be to copy and rename the entire plugin, this way they are 2 different plugins to Kong, yet have the same functionality. (untested!!)

**Comment by Efesto** [comment]
Is there currently any plan for allowing to apply a global scoped request-transformer together with a consumer-scoped?
